src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,362 @@
+import com.revrobotics.SparkMaxAbsoluteEncoder.Type;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import frc.robot.utils.SendableHelper;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.PrintCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotMap;
+import frc.robot.utils.AbsoluteEncoderChecker;
+import frc.robot.utils.ScoringLocationUtil;
+import frc.robot.utils.ScoringLocationUtil.ScoreHeight;
+
+public class Arm extends SubsystemBase {
+
+  public enum ArmPosition {
+    STARTING,
+    INTAKE,
+    SCORE_LOW,
+    SCORE_MID_HIGH,
+    AVOID_LIMELIGHT
+  }
+
+  public CANSparkMax armMotor =
+      new CANSparkMax(RobotMap.ARM_PIVOT_MOTOR_CAN_ID, MotorType.kBrushless);
+
+  private final RelativeEncoder armEncoder = armMotor.getEncoder();
+  private final AbsoluteEncoder armAbsoluteEncoder = armMotor.getAbsoluteEncoder(Type.kDutyCycle);
+
+  public ScoringLocationUtil scoreLoc;
+  private ArmPosition desiredPosition = ArmPosition.STARTING;
+  private ArmPosition latestPosition = ArmPosition.STARTING;
+  private ArmPosition goalPosition = ArmPosition.STARTING;
+  private boolean cancelScore = false;
+  private boolean scoringInProgress = false;
+
+  private AbsoluteEncoderChecker armAbsoluteEncoderChecker = new AbsoluteEncoderChecker();
+
+  TreeMap<ArmPosition, Double> armPositionMap;
+
+  /** Input deg, output Volts */
+  private ProfiledPIDController armController =
+  new ProfiledPIDController(
+      ArmCal.ARM_P,
+      ArmCal.ARM_I,
+      ArmCal.ARM_D,
+      new TrapezoidProfile.Constraints(
+          ArmCal.ARM_MAX_VELOCITY_DEG_PER_SECOND,
+          ArmCal.ARM_MAX_ACCELERATION_DEG_PER_SECOND_SQUARED));
+
+  public Arm(ScoringLocationUtil scoreLoc) {
+
+    armPositionMap = new TreeMap<ArmPosition, Double>();
+    armPositionMap.put(ArmPosition.STARTING, ArmCal.ARM_START_POSITION_DEG);
+    armPositionMap.put(ArmPosition.INTAKE, ArmCal.ARM_INTAKE_POSITION_DEG);
+    armPositionMap.put(ArmPosition.SCORE_LOW, ArmCal.ARM_LOW_POSITION_DEG);
+    armPositionMap.put(ArmPosition.SCORE_MID_HIGH, ArmCal.ARM_HIGH_MID_POSITION_DEG);
+    armPositionMap.put(
+        ArmPosition.AVOID_LIMELIGHT, ArmCal.ARM_AVOID_LIMELIGHT_POSITION_DEG);
+
+  this.scoreLoc = scoreLoc;
+  }
+
+  public void initialize() {}
+
+  @Override
+  // This method will be called once per scheduler run
+  public void periodic() {
+    // Check if position has updated
+    if (atPosition(ArmPosition.STARTING)) {
+      this.latestPosition = ArmPosition.STARTING;
+    } else if (atPosition(desiredPosition)) {
+      this.latestPosition = desiredPosition;
+    }
+
+    controlPosition(desiredPosition);
+    armAbsoluteEncoderChecker.addReading(armAbsoluteEncoder.getPosition());
```

Comment:
It doesn't seem like you actually use `armAbsoluteEncoderChecker`? You add readings but you don't use them.